PythonTool/TagExpansion.py

```diff
@@ -139,39 +119,79 @@ def tag_expansion(mesh_file, output_file = None):
 
     """
     For the vector tag on each element, retrieve the scalar value at a specific
-    index and create a scalar tag on the reference mesh. For each index,
-    write the scalar tag values to disk in a vtk file.
+    index and create a scalar tag. For each index, write the scalar tag value
+    to disk in a vtk file in the specified database.
     """
 
     index = 0
-    while index < reference_length:
+    while index < length:
         scalar_data = []
-        for tag in vec_tags_ext:
-            data = mb_ext.tag_get_data(tag, hexes_ext)
-            scalar_data = np.copy(data[:,index])
-            data_type = tag.get_data_type()
-            scalar_tag = mb_ref.tag_get_handle(tag.get_name(), 1, data_type,
-                                               types.MB_TAG_SPARSE, create_if_missing = True)
-            mb_ref.tag_set_data(scalar_tag, hexes_ref, scalar_data)
-
-        # Check if the user supplied a base output file name.
-        if(output_file):
-            file_location = os.getcwd() + "/" + dir_name + "/" + output_file + str(index) + ".vtk"
-        else:
-            file_location = os.getcwd() + "/" + dir_name + "/" + tag_name + str(index) + ".vtk"
+        data = mb_exp.tag_get_data(vec_tag, hexes)
+        scalar_data = np.copy(data[:,index])
+        data_type = vec_tag.get_data_type()
+        scalar_tag = mb_ref.tag_get_handle(name, 1, data_type, types.MB_TAG_SPARSE,
+                                           create_if_missing = True)
+        mb_ref.tag_set_data(scalar_tag, hexes, scalar_data)
+
+        # Write the mesh file with the new scalar tag.
+        file_location = os.getcwd() + "/" + dir_name + "/" + name + str(index) + ".vtk"
```

Comment:
Suggestion for a different file structure for writing out these files:
If you have a file `my_mesh.h5m` with vector tags `vec1` and `vec2`, then:

```
my_mesh_database / 
    vec1_database/
        my_mesh_vec1_<i>.h5m
    vec2_database/
        my_mesh_vec2_<j>.h5m
```
The final file names could be something different, but that makes it pretty specific while also keeping a cleaner directory system. Now if the script is run on the file more than once, the extra integer is just attached to the top level directory and represents the run number rather than the number of vector tags found.